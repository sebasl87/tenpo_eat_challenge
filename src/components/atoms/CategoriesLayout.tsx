import React from 'react'

import styled from 'styled-components/native';

export interface ICategoriesLayout {
    children: React.ReactNode;
    image?: any;
}

const LayoutImage = styled.ImageBackground`
    width: 174px;
    height: 80px;    
`
const TextCategories = styled.Text`
    color: #fff;
    position: absolute;
    z-index: 2;
    margin-top: 30px;
    font-size: 16px;
    letter-spacing: 1px;
`

const Container = styled.View`
    width: 168px;
    height: 90px;
    display: flex;
    align-items: center;    
    align-content: center;

`

const CategoriesLayout: React.FC<ICategoriesLayout> = ({ children, image }) => {
    return (
        <Container >
            <TextCategories style={{ fontFamily: "Gotham-Bold" }}>{children}</TextCategories>
            <LayoutImage source={image}>
            </LayoutImage>
        </Container>

    );
}

export default CategoriesLayout;