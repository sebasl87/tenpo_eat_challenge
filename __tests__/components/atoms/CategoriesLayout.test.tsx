import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import CategoriesLayout from "../../../src/components/atoms/CategoriesLayout";


describe("CategoriesLayout component", () => {
    it("should render successfully with CategoriesLayout", () => {
        const categoriesLayout = renderer
            .create(<CategoriesLayout children="AddressLayout_mock" />)
            .toJSON();

        expect(categoriesLayout).toMatchSnapshot();
    });
});
