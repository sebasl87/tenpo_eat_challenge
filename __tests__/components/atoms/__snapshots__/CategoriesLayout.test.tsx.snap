// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesLayout component should render successfully with CategoriesLayout 1`] = `
<View
  style={
    [
      {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "height": 90,
        "width": 168,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
          "letterSpacing": 1,
          "marginTop": 30,
          "position": "absolute",
          "zIndex": 2,
        },
        {
          "fontFamily": "Gotham-Bold",
        },
      ]
    }
  >
    AddressLayout_mock
  </Text>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      [
        {
          "height": 80,
          "width": 174,
        },
      ]
    }
  >
    <Image
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 80,
            "width": 174,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
